Issue/27 tdd pawn

owais34 opened the pull request:
Added the pawn class and pawn tests , added methods in board and abstract piece to reduce warnings.

programarivm commented:
Thanks for the PR, looks good! 

Let me just point out the following:

- It'd be convenient to TDD the `calcMobility()` method only; let's focus on other methods in upcoming issues.
- The friendly PR should be sent once the issue has been assigned.

Keep it up, and happy coding and learning.

owais34 commented:
Hi removed the redundant code from Board class , Removed redundant methods from Pawn and changed Pawn.test to test the calcMobility method only .
All 95 tests passing at present.
